src/blocking/delay.rs

```diff
@@ -12,15 +12,21 @@
 /// `UXX` denotes the range type of the delay time. `UXX` can be `u8`, `u16`, etc. A single type can
 /// implement this trait for different types of `UXX`.
 pub trait DelayMs<UXX> {
+    /// Enumeration of `DelayMs` errors
+    type Error;
+
     /// Pauses execution for `ms` milliseconds
-    fn delay_ms(&mut self, ms: UXX);
+    fn try_delay_ms(&mut self, ms: UXX) -> Result<(), Self::Error>;
```

Comment:
So, this is interesting. Can a busy-wait delay ever actually fail? If not, should this remain an infallible API?

Reply:
Good point. I also wondered about these traits. However, blocking behavior is not the same as busy-waiting.

Reply:
Could you elaborate on that last statement?

Reply:
Sure.
I understand blocking behavior as the fact that the further execution of the caller's code is stopped until the operation that was called has finished. i.e. the contract to the caller.
(In contrast to e.g. returning `Error::WouldBlock`).
I wanted to highlight that in the case of the delay this does not necessarily mean busy-waiting. For example, one could also think of an implementation à la cooperative multitasking which would still be an implementation which holds this contract (of course errors in this case need to be reported to the appropriate caller, it would be completely wrong to make the delay function fallible in order to return errors from another one, that is _not_ what I am saying).
The point being that maybe a different implementation which I do not know of exists for which it may make sense to fail.

Going back to my main statement, I made these methods fallible as well because the [meeting notes](https://hackmd.io/ck-xRXtMTmKYXdK5bEh82A?view#Final-Summary) explicitly stated:
> 3. For the next major release, we will ONLY be offering _Fallible_ traits, e.g. traits that return `Result`s. All trait method names will be prefixed with `try_` for consistency, e.g. `try_send`.

Nevertheless, I also wondered if failing would actually be possible for a delay implementation or if this was decided only for consistency (I was not at the meeting, maybe I should join some time).
It would be interesting to know if someone can think of a failing delay.

Reply:
> It would be interesting to know if someone can think of a failing delay.
I've given it some thought and although I can't think of one, I can't rule it out. This this is the correct approach.